src/strands/tools/registry.py

```diff
@@ -93,11 +96,20 @@ def add_tool(tool: Any) -> None:
                     if not function_tools:
                         logger.warning("tool_name=<%s>, module_path=<%s> | invalid agent tool", tool_name, module_path)
 
-            # Case 5: AgentTools (which also covers @tool)
+            # Case 5: ToolProvider
+            elif isinstance(tool, ToolProvider):
```

Comment:
When we introduce async-initialization (#330) we either need an async initialization of this method or we need the agent to do the tool loading - thoughts on if this should thus live in the agent instead of the `process_tools`?